temporalio/client.py

```diff
@@ -2827,6 +2829,8 @@ def __init__(
             # Use definition if callable
             if callable(workflow):
                 defn = temporalio.workflow._Definition.must_from_run_fn(workflow)
+                if not defn.name:
+                    raise TypeError("Cannot schedule dynamic workflow explicitly")
                 workflow = defn.name
             elif not isinstance(workflow, str):
```

Comment:
Can I just check, with the `TypeError`s above -- you're comfortable saying that the dynamic/non-dynamic distinction is a type distinction rather than a value distinction?

Reply:
Yeah, maybe I am using the wrong error. I was using type error to be like "wrong type of workflow passed in" but it is the value. I can change the error type to `ValueError`.

Reply:
Fixed